src/device/router_device_worker.erl

```diff
@@ -1869,12 +1885,28 @@ channel_correction_and_fopts(Packet, Region, Device, Frame, Count, ADRAdjustment
                 %% rate in the form of "SFdd?BWddd?" so that's what
                 %% we'll give it.
                 NewDr = lorawan_mac_region:dr_to_datar(Region, NewDataRateIdx),
+                NewTxPowerIdx = 0,
```

Comment:
You are setting `NewTxPowerIdx` to 0 here but it is also coming from above line 1879. Note that dialyzer is complaining because ` 0 =:= NewTxPowerIdx::pos_integer()`

Reply:
Fixed.